From: Jesse Barnes <jbarnes@hobbes.lan>
Date: Thu, 31 Jul 2008 20:07:20 +0000 (-0700)
Subject: Update DSPARB while planes are still off
X-Git-Tag: xf86-4_3_0_1
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/driver/xf86-video-intel.git;a=commitdiff;h=c0c63c9d8329d93288dc12a05fe0ba3cf71fd5af

Update DSPARB while planes are still off

This avoids the flickering people reported in the 2.4.0 release.
---

--- a/src/i830_display.c
+++ b/src/i830_display.c
@@ -1477,6 +1477,8 @@ i830_crtc_mode_set(xf86CrtcPtr crtc, Dis
     /* Wait for the clocks to stabilize. */
     usleep(150);
 
+    i830_update_dsparb(pScrn);
+
     OUTREG(htot_reg, (adjusted_mode->CrtcHDisplay - 1) |
 	((adjusted_mode->CrtcHTotal - 1) << 16));
     OUTREG(hblank_reg, (adjusted_mode->CrtcHBlankStart - 1) |
@@ -1510,8 +1512,6 @@ i830_crtc_mode_set(xf86CrtcPtr crtc, Dis
     
     i830WaitForVblank(pScrn);
 
-    i830_update_dsparb(pScrn);
-
     /* Clear any FIFO underrun status that may have occurred normally */
     OUTREG(pipestat_reg, INREG(pipestat_reg) | FIFO_UNDERRUN);
 }
